src/GapicClientTrait.php

```diff
@@ -195,44 +201,31 @@ private function buildClientOptions(array $options)
             $options['transportConfig']['rest'] += $defaultOptions['transportConfig']['rest'];
         }
 
-        $this->modifyClientOptions($options);
+        // These calls do not apply to "New Surface" clients.
+        if (!$this->isNewClientSurface()) {
+            $preModifiedOptions = $options;
+            $this->modifyClientOptions($options);
+            // NOTE: this is required to ensure backwards compatiblity with $options['apiEndpoint']
+            if ($options['apiEndpoint'] !== $preModifiedOptions['apiEndpoint']) {
+                $apiEndpoint = $options['apiEndpoint'];
+            }
 
-        // serviceAddress is now deprecated and acts as an alias for apiEndpoint
-        if (isset($options['serviceAddress'])) {
-            $options['apiEndpoint'] = $this->pluck('serviceAddress', $options, false);
+            // serviceAddress is now deprecated and acts as an alias for apiEndpoint
+            if (isset($options['serviceAddress'])) {
+                $apiEndpoint = $this->pluck('serviceAddress', $options, false);
+            }
```

Comment:
It seems the order is incorrect. The `$apiEndpoint` should first look at `serviceAddress` from options and then the `apiEndpoint` not the other way around.

Reply:
I agree, but unfortunately, the way we have it implemented is _currently_ in this order, and changing that could potentially break the existing behavior.